feature_engine/timeseries/forecasting/window_features.py

```diff
@@ -0,0 +1,270 @@
+    WindowFeatures adds new features to a dataframe based on window operations. Window
+    operations are operations that perform an aggregation over a sliding partition of
+    past values. A window feature is, in other words, a feature created after computing
+    statistics (e.g., mean, min, max, etc.) using a window over the past data. For
+    example, the mean value of the previous 3 months of data is a window feature. The
+    maximum value of the previous three rows of data is another window feature.
+
+    WindowFeatures uses the pandas' functions `rolling()`, `agg` and `shift`. With
+    `rolling`, it creates rolling windows. With `agg` it applies multiple functions
+    within those windows. With 'shift()' it allocates the values to the correct rows.
+
+    For supported aggregation functions, see Rolling Window
+    `Functions <https://pandas.pydata.org/docs/reference/window.html>`_.
+
+    With pandas `rolling` we can perform rolling operations over 1 window size at a
+    time. WindowFeatures builds on top of pandas `rolling()` in that multiple windows
+    can be evaluated at the same time and the created features will be concatenated to
+    the original dataframe.
+
+    To be compatible with WindowFeatures, the dataframe's index must have unique values
+    and no NaN.
+
+    WindowFeatures works only with numerical variables. You can pass a list of variables
+    to use as input for the windows. Alternatively, WindowFeatures will automatically
+    select all numerical variables in the training set.
+
+    More details in the :ref:`User Guide <window_features>`.
+
+    Parameters
+    ----------
+    {variables}
+
+    window: int, offset, BaseIndexer subclass, or list, default=3
+        Size of the moving window. If an integer, the fixed number of observations used
+        for each window. If an offset, the time period of each window. If list, features
+        will be created for each one of the windows in the list. It can also take a
+        function. See parameter `windows` in the pandas `rolling()` documentation for
+        more details.
+
+    min_periods: int, default None.
+        Minimum number of observations in window required to have a value; otherwise,
+        result is np.nan. See parameter `min_periods` in the pandas `rolling()`
+        documentation for more details.
+
+    functions: list of strings, default = ['mean']
+        The functions to apply within the window. Valid functions can be found
+        `here <https://pandas.pydata.org/docs/reference/window.html>`_.
+
+    periods: int, list of ints, default=1
+        Number of periods to shift. Can be a positive integer. See param `periods` in
+        pandas `shift`.
+
+    freq: str, list of str, default=None
+        Offset to use from the tseries module or time rule. See parameter `freq` in
+        pandas `shift()`.
+
+    sort_index: bool, default=True
+        Whether to order the index of the dataframe before creating the lag features.
```

Comment:
Typo: I think it should say `... window features.` rather than `... lag features.`